app/models/searchable_record.rb

```diff
@@ -0,0 +1,18 @@
+# The base class for all searchable models
+class SearchableRecord < ApplicationRecord
+  self.abstract_class = true
+
+  def self.searchable_attributes
+    []
```

Comment:
This is more of a comprehension question, but why the `def self.method` instead of `def method` when other classes don't use the self?

Reply:
self.method means this can only be called to the Class and not to an instance. That means you would use SearchableRecord.seachable_attributes but not SearchableRecord.new().seachable_attributes. This is important to not save the list in the datebase, instead only once in the class.

Reply:
Okay, thanks for the explanation 👍🏻